### PR TITLE
DEV/MNT: migrate to GitHub Actions and pyproject.toml [LCLSPC-603]

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Upgrading by way of https://github.com/pcdshub/pcds-python-migration-tools
* lucid no longer uses Travis CI and has migrated to GitHub Actions for continuous integration, testing, and documentation deployment.
* lucid has been migrated to use setuptools-scm, replacing versioneer, as its version-string management tool of choice.
* lucid has been migrated to use the modern ``pyproject.toml``, replacing ``setup.py`` and related files.
* Older language features and syntax found in the repository have been updated to Python 3.9+ standards by way of ``pyupgrade``.
* Sphinx 6.0 is now supported for documentation building. ``docs-versions-menu`` replaces ``doctr-versions-menu`` and ``doctr`` usage for documentation deployment on GitHub Actions.  The deployment key is now no longer required.
* `pyupgrade` was used to update annotations, though there might not have been many here
* `pycln` was used to clean up imports

## Motivation and Context
* Migrating because we want to migrate our projects from Travis CI for one

## How Has This Been Tested?
* GitHub Actions

## Where Has This Been Documented?
This PR and Jira [LCLSPC-603](https://jira.slac.stanford.edu/browse/LCLSPC-603)

### CODE REVIEW

**Comment by klauer** [comment]
I think for the purposes of testing on pip/conda and building the docs without failing, I'd like to make pyqtads an optional dependency. 

I don't think it'll be much work. What do you think @ZLLentz @tangkong ?

**Comment by ZLLentz** [comment]
It's not obvious to me how to make pyqtads optional since it's used for the docking. I guess there'd just be an optional code path with no docking support?

**Comment by klauer** [comment]
Yes, exactly @ZLLentz. It'd just be creating independent windows that can't be docked.

It's either that or we just live with most of our CI jobs failing.

**Comment by ZLLentz** [comment]
It sounds kind of annoying to implement to me but maybe I haven't looked into it enough. It's a good idea if you think it's not too hard.

**Comment by klauer** [comment]
Gave up on making pyqtads optional here, and as such I set pypi-backed jobs to be optional for PR CI.
I think this one should be good to go

**Review by tangkong — APPROVED** [review]
I think this looks good, a shame the pyqtads stuff didn't pan out, but otherwise this seems standard.

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -1,56 +1,55 @@
...
   dev_url: https://github.com/pcdshub/lucid
-  summary: 'SLAC National Accelerator Laboratory: LCLS User Control and Interface Design'
+  home: https://github.com/pcdshub/lucid
   license: SLAC Open
   license_family: OTHER
   license_file: LICENSE.md
+  summary: 'SLAC National Accelerator Laboratory: LCLS User Control and Interface Design'
 
 extra:
   recipe-maintainers:
```
Comment (ZLLentz) [inline conda-recipe/meta.yaml:52]: Do our other projects have a recipe-maintainers section in their self-contained (non-forge) recipes? I don't remember seeing this outside of conda-forge.

**Review by ZLLentz — APPROVED** [review]

File: conda-recipe/meta.yaml (hunk 1 above)
Comment (klauer) [reply conda-recipe/meta.yaml:52]: Not typically, no - this probably could have been cleaned

Comment (klauer) [reply conda-recipe/meta.yaml]: Removed in 29c75d1

**Pull request merged**